packages/@ourworldindata/grapher/src/slopeCharts/SlopeChart.tsx

```diff
@@ -928,7 +973,11 @@ class LabelledSlopes
                     s1 !== s2 &&
                     s1.hasLeftLabel &&
```

Comment:
Since we always show the right label but only sometimes the left one, we should be doing collision detection on the right side first.

This will get rid of the current "jumping" behaviour, where hovering over some labels can change which labels are displayed _way below/above_ that label. See the gif below, e.g. Argentina/Botswana.

![CleanShot 2024-03-11 at 10 21 30](https://github.com/owid/owid-grapher/assets/2641501/143b9f3e-6a03-430c-9e89-c223bb0e7cca)

In making this change, note that in `chooseLabels`, you should also replace the check for `hasLeftLabel` with `hasRightLabel`.

Reply:
Good call!